libtheora: fix how package is configured

hppritcha opened the pull request:
Not sure how this was ever working before, but here's what I'm seeing trying to build paraview, which now has a dependency on libtheora:
```
==> libtheora: Executing phase: 'autoreconf'
==> libtheora: Executing phase: 'configure'
==> Error: ProcessError: Command exited with status 1:
    '/tmp/hpp/spack-stage/spack-stage-libtheora-1.1.1-4ujjd67rwaly4ca36pf2pfnex6p4bokx/spack-src/configure' '--prefix=/users/hpp/spack/opt/spack/cray-rhel8-aarch64/gcc-8.3.1/libtheora-1.1.1-4ujjd67rwaly4ca36pf2pfnex6p4bokx'

1 error found in build log:
     225    checking for a BSD-compatible install... /usr/bin/install -c
     226    checking whether build environment is sane... yes
     227    checking for a thread-safe mkdir -p... /usr/bin/mkdir -p
     228    checking for gawk... gawk
     229    checking whether make sets $(MAKE)... yes
     230    checking whether make supports nested variables... yes
  >> 231    configure: error: source directory already configured; run "make distclean" there first
```
The problem is the autogen.sh script in the package also runs configure.  So when spack goes ahead and tries to run a configure stage, things don't work.

This commit reworks the build process to avoid this unneeded configure step

Signed-off-by: Howard Pritchard <howardp@lanl.gov>

hppritcha commented:
I'll have subsequent PR to fix the build itself.

hppritcha commented:
i've no idea what the PGP error is about with unittests (3.8)

alalazo commented:
@hppritcha Restarted the tests, I believe you hit https://github.com/spack/spack/issues/17405

adamjstewart commented:
Instead of overriding `phases`, could we instead override `configure` to be a no-op? Something like:
```python
def configure(self, spec, prefix):
    # The autogen.sh script already runs configure
    pass
```

adamjstewart commented:
Alternatively, we could patch `autogen.sh` to not run `configure`...

hppritcha commented:
@adamjstewart i'll try out the autogen.sh patching idea.

hppritcha committed:
libtheora: patch autogen.sh to avoid

configure problem when spack runs the configure stage of its build/install pipeline

Signed-off-by: Howard Pritchard <howardp@lanl.gov>

hppritcha commented:
@adamjstewart check this PR now.  Patch file approach looks good to me.

adamjstewart commented:
Looks good to me. I would encourage you to open an issue with the developers. `autogen.sh` shouldn't run `configure`, it should only generate it.

hppritcha commented:
@adamjstewart could we get this merged sometime soon?  its impeding progress with one of our production application's dependency build infrastructure standup.
